### ISSUE TITLE
Dredd crashes if Gavel throws an error

### ISSUE THREAD

**Issue by honzajavorek** [apiaryio/dredd#731]
Gavel.js can throw an error, e.g. in case it's provided with an invalid media type like this:

```apib
+ Response 200 (image)
```

Dredd crashes with following output:

```
debug: Gavel.js validation errored: TypeError: invalid media type
    at splitType (/Users/honzajavorek/Workspace/dredd-example/node_modules/media-typer/index.js:249:11)
    at Object.parse (/Users/honzajavorek/Workspace/dredd-example/node_modules/media-typer/index.js:160:13)
    at HttpResponse.Validatable.isJsonContentType (/Users/honzajavorek/Workspace/dredd-example/node_modules/gavel/lib/mixins/validatable-http-message.js:356:27)
    ...
TypeError: this.emitError is not a function
    at /Users/honzajavorek/Workspace/dredd-example/node_modules/dredd/lib/transaction-runner.js:638:14
    at proxy (/Users/honzajavorek/Workspace/dredd-example/node_modules/gavel/lib/validate.js:16:12)
    at Object.isValid (/Users/honzajavorek/Workspace/dredd-example/node_modules/gavel/lib/validate.js:32:10)
    ...
```

Note output like this is given only with `--level=debug`, without it it's even worse. Dredd should be able to recover, provide a nice human-readable error message, gracefully abort testing and exit with status 1.

**Comment by austin-a** [comment]
I'm trying to use Dredd for a project and I'm encountering a similar error. Gavel.js seems to be crashing in the same location as your example, but I'm not using an image.

`verbose: About to perform an HTTPS request to the server under test: PUT <pathtoAPI>
verbose: Handling HTTP response from tested server
verbose: Running 'beforeEachValidation' hooks
debug: Running hooks...
verbose: Running 'beforeValidation' hooks
verbose: Validating HTTP transaction by Gavel.js
debug: Determining whether HTTP transaction is valid (getting boolean verdict)debug: Gavel.js validation errored: TypeError: argument string is required
    at Object.parse (C:\Users\AAmort\AppData\Roaming\npm\node_modules\dredd\node_modules\media-typer\index.js:141:11)`

`TypeError: this.emitError is not a function
    at C:\Users\AAmort\AppData\Roaming\npm\node_modules\dredd\lib\transaction-runner.js:634:14
`

**Comment by austin-a** [comment]
```
put:
      tags:
        - ServiceQuotes
consumes:
        - application/json
        - application/xml
      produces:
        - application/json
        - application/xml
...
parameters:
        - name: serviceQuoteStatus
          in: body
          required: true
          schema:
            maxItems: 1
            $ref: '#/definitions/serviceQuoteStatus'
          description: ''
        - name: id
          in: path
          required: true
          description: ''
          type: string
          x-example: '123456'
```
The schema reference is defined here:
```
serviceQuoteStatus:
    type: object
    properties:
      accountNumber:
        type: string
      salesChannel:
        type: string
      status:
        type: string
    default: '"status": "Cancel"'
```

This is for a put request, at first I was afraid that I was sending an empty request for a field that was required, so I tried to setup a default value for the serviceQuoteStatus object, but now I'm wondering if it is a problem with the expected response. The expected response is a 204 with no content. Perhaps my problem is related to [#556](https://github.com/apiaryio/dredd/issues/556).

**Comment by honzajavorek** [comment]
It could be related. How does the expected response look like in your API description document? I can see only the request from your snippets.

**Comment by austin-a** [comment]
```
responses:
    '204':
        description: No Content
```

**Comment by honzajavorek** [comment]
I think this is not tested very well at the moment 😬 This needs to be fixed. I think it is related to #556.

**Comment by honzajavorek** [comment]
Fixed by https://github.com/apiaryio/dredd/pull/772. @austin-a your issue needs to be resolved together with #556, but you should experience a better error reporting with v3.4.3.
